src/game.ts

```diff
@@ -0,0 +1,147 @@
+      try {
+        const boardCoordinates = getBoardCoordinatesFromUserInput(userInput);
+        currentState = updateGameState(currentState, boardCoordinates);
+        isUserCoordinatesAreValid = true;
+      } catch (error) {
+        console.error(error.message);
+      }
+    }
+    displayGameState(currentState);
+  }
+}
+
+function askCellCoordinatesToUser(): Promise<string> {
+  const readline = createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+  return new Promise((resolve) => {
+    readline.question(
+      "Where would you like to put you stone ?",
+      (cellCoordinates) => {
+        readline.close();
+        resolve(cellCoordinates);
+      }
+    );
+  });
+}
+
+function getBoardCoordinatesFromUserInput(
+  userCoordinates: string
+): Coordinates {
+  let x: number, y: number;
+  if (userCoordinates.length < 2) {
+    throw new Error(
+      `Given coordinates is not compatible. Must be something like "H4"`
+    );
+  }
+  if (/[A-Z]/.test(userCoordinates[0])) {
+    x = userCoordinates[0].charCodeAt(0) - UTF16_CODE_OF_LETTER_A;
+  } else {
+    throw new Error(`Given character is not included in the range [A,Z]`);
+  }
+  y = parseInt(userCoordinates.slice(1, userCoordinates.length));
+  if (y >= 0 && y <= 25) {
+    return { x: x, y: y };
+  } else {
+    throw new Error(`Given number is not included in the range of [0,25]`);
+  }
+}
+
+export function generateNewBoard(): GameState {
+  return {
+    board: [
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+    ],
+  };
+}
+
+function updateGameState(
+  previousState: GameState,
+  actionCoordinates: Coordinates
+): GameState {
+  if (!areCoordinatesValid(previousState, actionCoordinates)) {
+    throw new Error("Given coordinates are outside the scope of the board.");
+  }
+  if (
+    previousState.board[actionCoordinates.y][actionCoordinates.x].value !==
+    "empty"
+  ) {
+    throw new Error("A pawn is already set in the selected cell.");
+  }
+  const newGameState = { board: previousState.board };
+  newGameState.board[actionCoordinates.y][actionCoordinates.x].value = "black";
+  return newGameState;
+}
+
+export function areCoordinatesValid(
+  gameState: GameState,
+  coordinates: Coordinates
+): boolean {
+  if (coordinates.y < 0 || coordinates.y >= gameState.board.length) {
+    return false;
+  }
+  if (
+    coordinates.x < 0 ||
+    coordinates.x >= gameState.board[coordinates.y].length
+  ) {
+    return false;
+  }
+  return true;
```

Comment:
You can ease this function according to the name of this function. You are checking that the coordinates are valid so check only that return y inside the board && x inside the board.